src/components/table/template.test.js

```diff
@@ -10,589 +10,506 @@ const { render, getExamples } = require('../../../lib/jest-helpers')
-    expect($firstTableHeader.hasClass('govuk-table__header')).toBeTruthy()
 
-    const $lastTableRow = $tableBody.find('.govuk-table__row:last-child')
-    const $lastTableHeader = $lastTableRow.find('.govuk-table__header:first-child')
+    it('can have additional classes', () => {
+      const $ = render('table', examples['table with head and caption'])
+      const $caption = $('.govuk-table__caption')
 
-    expect($lastTableHeader.get(0).tagName).toEqual('th')
-    expect($lastTableHeader.attr('scope')).toEqual('row')
-    expect($lastTableHeader.hasClass('govuk-table__header')).toBeTruthy()
+      expect($caption.hasClass('govuk-heading-m')).toBeTruthy()
+    })
   })
 
-  it('includes additional classes when the first row is a header', () => {
-    const $ = render('table', {
-      'firstCellIsHeader': true,
-      'rows': [
-        [
-          {
-            'text': 'January',
-            'classes': 'my-custom-class'
-          },
+  // =========================================================
+  // Column headers
+  // =========================================================
+
+  describe('column headers', () => {
+    it('can be specified', () => {
+      const args = examples['table with head']
+      const $ = render('table', args)
+
+      const headings = $('.govuk-table').find('thead tr th')
+        .map((_, e) => $(e).text())
+        .get()
+
+      expect(headings).toEqual([
+        'Month you apply',
+        'Rate for bicycles',
+        'Rate for vehicles'
+      ])
+    })
+
+    it('have HTML escaped when passed as text', () => {
+      const $ = render('table', {
+        head: [
           {
-            'text': '£85',
-            'format': 'numeric'
+            text: 'Foo <script>hacking.do(1337)</script>'
           }
         ]
-      ]
-    })
+      })
 
-    expect($('.govuk-table__row *:first-child').hasClass('my-custom-class')).toBeTruthy()
-  })
+      const $th = $('.govuk-table thead tr th')
 
-  it('includes rowspan when the first row is a header', () => {
-    const $ = render('table', {
-      'firstCellIsHeader': true,
-      'rows': [
-        [
-          {
-            'text': 'January',
-            'rowspan': 2
-          },
+      expect($th.text()).toEqual('Foo <script>hacking.do(1337)</script>')
+    })
+
+    it('allow HTML when passed as HTML', () => {
+      const $ = render('table', {
+        head: [
           {
-            'text': '£85',
-            'format': 'numeric'
+            html: 'Foo <span>bar</span>'
           }
         ]
-      ]
-    })
+      })
 
-    expect($('.govuk-table__row *:first-child').attr('rowspan')).toEqual('2')
-  })
+      const $th = $('.govuk-table thead tr th')
 
-  it('includes colspan when the first row is a header', () => {
-    const $ = render('table', {
-      'firstCellIsHeader': true,
-      'rows': [
-        [
-          {
-            'text': 'January',
-            'colspan': 2
-          },
+      expect($th.text()).toEqual('Foo bar')
```

Comment:
I think these 'work' in that they would correctly fail if the escaping logic was incorrect:

If the text was being escaped, `$th.text()` would be equal to `Foo <span>bar</span>` so the test would fail.

In the other example, if the escaping wasn't working then `$th.text`() would be equal to just 'Foo', so the test would fail.

Do you think asserting against the HTML is more explicit?

Reply:
It was my understanding that `.text()` would get all the text within any markup.

So this test isn't great as you could change the inputs and it'd still pass.

Not blocking, just feels like it should be more explicit.